### ISSUE TITLE
IDE0059 - Unneccessary assignment is not reported if lambda is used in method

### ISSUE THREAD

**Issue by Dreamescaper** [dotnet/roslyn#38376]
**Version Used**: 
VS16.3 Preview 2

**Steps to Reproduce**:

```csharp
using System;

namespace TestNamespace
{
    public class TestClass
    {
        public void TestMethod()
        {
            var r = new Random();
            DoSmth(() => true);
        }

        private void DoSmth(Func<bool> func) => func();
    }
}
```

**Expected Behavior**:
IDE0059 should be triggered for `r`.

**Actual Behavior**:
IDE0059 is not triggered. 
If you comment out `DoSmth(() => true);`, IDE0059 is triggered as expected.
